models/transaction_test.go

```diff
@@ -13,10 +14,36 @@ import (
 	"github.com/stretchr/testify/require"
 )
 
-func createTestEvent(t *testing.T) (cadence.Event, *gethTypes.Transaction, *types.Result) {
+//go:embed fixtures/evm_tx.bin
+var evmTxBinary string
+
+//go:embed fixtures/direct_call.bin
+var directCallBinary string
+
+func createTestEvent(t *testing.T, txBinary string) (cadence.Event, *types.Result) {
+	txEncoded, err := hex.DecodeString(txBinary)
+	require.NoError(t, err)
+
+	var txHash common.Hash
+	var txType uint8
+	if txEncoded[0] == types.DirectCallTxType {
+		directCall, err := types.DirectCallFromEncoded(txEncoded)
+		require.NoError(t, err)
+
+		txHash, err = directCall.Hash()
+		require.NoError(t, err)
+
+		txType = types.DirectCallTxType
+	} else {
+		gethTx := &gethTypes.Transaction{}
+		gethTx.UnmarshalBinary(txEncoded)
+		txHash = gethTx.Hash()
+		txType = gethTx.Type()
+	}
```

Comment:
Because the `cadence.Event` created in this helper function, is actually fed as input to `DecodeTransaction` in the following test cases :sweat_smile: